apps/dashboard/app/controllers/launchers_controller.rb

```diff
@@ -94,7 +94,15 @@ def submit_script_params
   end
 
   def save_script_params
-    params.permit({ launcher: SAVE_SCRIPT_KEYS }, :project_id, :id)
+    auto_env_params = params[:launcher].keys.select { |k| k.match?('auto_environment_variable') }
+
+    auto_env_params.each do |p|
+      params[:launcher][p] = params[:launcher][p].match(/.*=(.*)/)[1]
+    end
+
+    whitelisted_params = SAVE_SCRIPT_KEYS + auto_env_params
+
+    params.permit({ launcher: whitelisted_params }, :project_id, :id)
```

Comment:
I think this is the bit where I'm talking about simplifying so that the controller ever sees because the value `FOO` is already embedded in the key.

```
launcher[auto_environment_variable_FOO]=BAR
```

instead of 
```
launcher[auto_environment_variable_FOO]="FOO=BAR"
```